### ISSUE TITLE
Broken Images in pump 4.x and higher

### ISSUE THREAD

**Issue by vxcamiloxv** [pump-io/pump.io#1435]
After test in pump 4.x and 5.x, I upload image and get 403 error in logs when go to the post, I try to visit image url and get the same error, I test in pump 3.0.1 (same version of identi.ca) and work fine

[Looking in the code](https://github.com/pump-io/pump.io/blob/master/routes/uploads.js#L68) the `req.principal` object is `undefined`, probably is the error cause

**Comment by jankusanagi** [comment]
For the record, I've been using servers with 4.x for a while, and 5.0.0 now, and I haven't seen this issue, so it's probably a configuration/permissions issue on your part.

Did you do the migrations carefully, especially 2→3, the uploadDir/datadir thing?
https://pumpio.readthedocs.io/en/latest/upgrades/2.x-to-3.x.html
https://pumpio.readthedocs.io/en/latest/upgrades/3.x-to-4.x.html
https://pumpio.readthedocs.io/en/latest/upgrades/4.x-to-5.x.html

**Comment by strugee** [comment]
Yeah, this has been working for me for a while. Post your `pump.io.json`?

**Comment by vxcamiloxv** [comment]
I thought it was permissions errors but not:

Initial test in 5.x (master) -> error

```
git checkout v4.0.0
rm -r ./node_modules
npm install
```

Test over 4.0.0 -> same error

```

git checkout v3.0.1
rm -r node_modules
npm install 
````

Test over 3.0.1 -> show images

mi config: 


```
{
    "driver":  "redis",
    "params":  {"database": "2"},
    "secret":  "my great secret",
    "noweb":  false,
    "site":  "Test Pump",
    "owner":  "vxcamiloxv",
    "ownerURL":  "http://example.com/",
    "port":  31337,
    "hostname":  "localhost",
    "address":  "localhost",
    "nologger":  false,
    "children": 1,
    "datadir": "/home/vxcamiloxv/tmp",
    "enableUploads": true,
    "debugClient": false,
    "noCDN": true
}
```

**Comment by vxcamiloxv** [comment]
Parabola GNU/Linux - node v8.1.1 - Firefox 55

**Comment by vxcamiloxv** [comment]
The [PR #1438](https://github.com/pump-io/pump.io/pull/1438) fixes this issue

**Comment by vxcamiloxv** [comment]
You can reproduce this error in datamost.com

![image](https://user-images.githubusercontent.com/14242544/31629608-ce99b86a-b279-11e7-9a36-f464c0415b78.png)

**Comment by strugee** [comment]
> only for non-public images

This explains everything. For public images it wouldn't matter that there was no session since it's, well, public. I always post everything public which explains why I couldn't reproduce this too.

**Comment by strugee** [comment]
Reopening as I'd like to backport this to stable versions, when I get a working computer... *sigh*

Probably we should just apply this patch on top of the 5.1 release
